converter/src/main/java/gov/cms/qpp/conversion/model/Node.java

```diff
@@ -315,16 +318,23 @@ public void setDefaultNsUri(String newDefaultNsUri) {
 	 */
 	private List<Node> findNode(TemplateId templateId, Predicate<List<Node>> bail) {
 		List<Node> foundNodes = new ArrayList<>();
-		if (this.type == templateId) {
-			foundNodes.add(this);
-		}
-		for (Node childNode : childNodes) {
+		List<Node> toSearch = Lists.newArrayList(childNodes);
+		Consumer<Node> templateCheck = node -> {
+			if (node.getType() == templateId) {
+				foundNodes.add(node);
+			}
+		};
+		templateCheck.accept(this);
+
+		for (int i = 0; i < toSearch.size(); i++) {
```

Comment:
Oh, interesting.  Was the issue that Java never went into the childrens' child nodes since we were using iteration (the `for (Node childNode : childNodes)` form)?

Reply:
No, the prior version recursed through respective child node's hierarchy before advancing through it's siblings. This change adds iterated node's children to the same list the loop is iterating giving the effect of generational iteration.